### PR TITLE
handle nested calls's err

### PR DESCRIPTION
**Pull request by mask-pp**
Record err in `CREATE, CREATE2, CALL, CALLCODE, DELEGATECALL, STATICCALL` opcodes.

### CODE REVIEW

File: core/vm/logger.go [hunk 1]
```diff
@@ -334,10 +334,11 @@ func (l *StructLogger) CaptureExit(output []byte, gasUsed uint64, err error) {
 
 	theLogPos := l.callStackLogInd[stackH-1]
 	l.callStackLogInd = l.callStackLogInd[:stackH-1]
-	theLog := l.logs[theLogPos]
+	theLog := &l.logs[theLogPos]
 	// update "forecast" data
 	if err != nil {
 		theLog.ExtraData.CallFailed = true
+		theLog.Err = err
```
Comment (noel2004) [inline core/vm/logger.go:341]: I think theLog should has a `*StructLogRes` type, which owns `Error` field rather than `Err`?

**Review by noel2004 — COMMENTED** [review]
And I think in `CaptureFault` we should also set the `Error` field for the *LAST* oplog under following conditions :

1.  the  passed `err` argument is not nil
2.  the `Depth` in op log is equal to the passed `depth` argument

**Comment by noel2004** [comment]
Also, add `Error` field in `ExecutionResult` and set it from `CaptureEnd`

**Comment by mask-pp** [comment]
> 

Why not catch err directly in Call opcode?

**Comment by mask-pp** [comment]
> And I think in `CaptureFault` we should also set the `Error` field for the _LAST_ oplog under following conditions :
> 
> 1. the  passed `err` argument is not nil
> 2. the `Depth` in op log is equal to the passed `depth` argument

Catch err in `CaptureExit` is enough.

**Comment by mask-pp** [comment]
> 

Is it really necessary to do it? Is it enough to record err message in opcodes' call path?

**Comment by Thegaram** [comment]
Closing after confirmation from @mask-pp.

**Pull request closed without merging**